conjureup/controllers/newcloud/gui.py

```diff
@@ -70,32 +70,22 @@ def finish(self, credentials=None, back=False):
 
         region = None
         if credentials is not None:
-            common.save_creds(app.current_cloud, credentials)
             cloud_type = juju.get_cloud_types_by_name()[app.current_cloud]
-            if cloud_type == 'maas':
-                # when bootstrapping a MAAS controller that hasn't been added
-                # as a saved cloud, we have to provide the API endpoint as
-                # the "region" when bootstrapping
-                region = credentials.fields()[0].value
-
-            # XXX: Oracle is handled special case until that cloud is
-            # solidified and makes it into Juju's default cloud listing
-            if cloud_type == 'oracle':
-                # need to create a custom cloud here with endpoint for oracle
-                endpoint = credentials.fields()[3].value
-                oracle_config = {
-                    'type': 'oracle',
-                            'description': 'Oracle Cloud',
-                            'auth-types': ['userpass'],
-                            'endpoint': endpoint,
-                            'regions': {
-                                'uscom-central-1': {}
-                            }
-                }
+            if cloud_type == 'maas' or cloud_type == 'oracle':
```

Comment:
Seeing this made me wonder what would happen if you selected an existing cloud of one of these types.  Following that code path, I noticed [newcloud/gui.py#L38](https://github.com/conjure-up/conjure-up/blob/master/conjureup/controllers/newcloud/gui.py#L38) and now I'm wondering why we're bothering to check the endpoint at all instead of just checking whether `juju.get_cloud(app.current_cloud)` returns something?  In particular, it seems like there's an edge case there where a given cloud type has existing credentials saved but no actual cloud (or the user is adding a second cloud of that type). **Edit:** I just realized that the only clouds we allow adding are the special-case ones, lxd (which follows an entirely separate branch), and maas & oracle (which both have a top-level `endpoint`, unlike all of the predefined clouds).  So I guess this concern is moot.

As for my original question, I think the flow here is ok, but I do think it might be easier to follow if the "register a new cloud" action was an entirely separate callback from "select an existing cloud" callback.  A refactor for another day, perhaps, unless you agree and feel like doing it as part of this.